### ISSUE TITLE
fix(youtu_vl): vision processor emits patches in the wrong order

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1610]
## Problem / Background

`YoutuVLProcessor::try_preprocess_with_spatial` in `src/vision/processors/youtu_vl.rs` emits one row per patch in plain raster order over the patch grid, with the inner feature order `(c, dy, dx)`:

```rust
for patch_idx in 0..total_patches_img {
    let py = patch_idx / w_patches as usize;
    let px = patch_idx % w_patches as usize;
    // ...
    for c in 0..in_channels {
        for dy in 0..self.patch_size {
            for dx in 0..self.patch_size {
```

Two independent consumers expect a different order, so the vision tower is fed patches that do not correspond to the positions and groupings it assumes.

**1. The checkpoint's own processor.** `image_processing_siglip2_fast.py::convert_image_to_patches`, called at line 303 with `merge_size=2`, reshapes to `(C, nh/m, m, ps, nw/m, m, ps)` and permutes `(1, 4, 2, 5, 3, 6, 0)`. That yields rows ordered `(block_row, block_col, mh, mw)`, which is merge-block-major, and inner features `(dy, dx, c)`, which is channel-last. `Siglip2VisionEmbeddings.patch_embedding` is an `nn.Linear(num_channels * patch_size * patch_size, embed_dim)` applied directly to those rows, and `remap_youtu_vl_weights` in `src/loading/vlm_youtu_vl.rs` only renames that weight; it never permutes it. So both the row order and the inner feature order differ from what the loaded weight was trained against.

**2. mlxcel's own encoder.** `YoutuVLVisionEncoder::rot_pos_emb` in `src/vision/encoders/youtu_vl.rs` builds the position ids with `reshape(&[h / merge, merge, w / merge, merge])` followed by `transpose_axes(&[0, 2, 1, 3])`, which is exactly merge-block-major. `forward_with_spatial` then reshapes the hidden states to `[n_groups, spatial_merge_unit, dim]` and gathers along the group axis, which assumes each consecutive group of `spatial_merge_unit` rows is one 2x2 patch block.

The mismatch is therefore internal and provable without any external oracle: the encoder's two inputs disagree. Each token receives the rotary position of a different spatial location, and the patch merger combines four patches that are not a 2x2 block.

The repository's own Qwen2-VL processor does this correctly (`src/vision/processors/qwen2_vl.rs:377-382`), and has a test named `owned_and_mlx_paths_share_spatial_merge_grouped_patch_order` pinning it. The Youtu-VL processor reads as a copy of that one which dropped the block loop and kept Qwen2-VL's channel-first inner order. The stale comment "with the inner ordering `(c, dy, dx)` to match how upstream unfolds patches via `unfold`" is the fingerprint: Qwen2-VL unfolds, and this checkpoint does not.

## Evidence

Measured on `models/mlx/youtu-vl-4b-instruct` (`tencent/Youtu-VL-4B-Instruct`), M1 Ultra, release build with `metal,accelerate`, greedy (`-t 0 -n 48`).

| fixture | what it is | mlxcel says |
|---|---|---|
| `tests/fixtures/test_image.png` | a 224x224 solid **orange** square | "The image is completely black and contains no visible content, objects, text, or details. It is a solid black square or rectangle with no variation in color or texture." |
| `tests/fixtures/test_image_shapes.png` | 448x448, a red square, a blue circle and a green triangle on light grey | "The image contains a single, solid black circle on a white background." |
| `tests/fixtures/test_image_shapes_336.png` | the same scene at 336x336 | "The image contains a single, solid black circle on a white background." |

The 224x224 row isolates this defect from the window-permutation defect fixed in #1600. At that size the merged grid is 7x7, which is a single 8x8 window, so `get_window_index` is the identity and the window inverse provably cannot change the result. The description is still wrong, so the corruption is upstream of the window reordering.

The cause was then confirmed directly. Patching only the emission loop to the upstream ordering below, rebuilding, and rerunning the same three commands:

| fixture | raster order, `(c, dy, dx)` | merge-block-major, `(dy, dx, c)` |
|---|---|---|
| `test_image.png` (solid orange) | "The image is completely black ... a solid black square or rectangle" | "The image is a solid, uniform block of **bright orange** color. There are no visible objects, textures, patterns, or variations" |
| `test_image_shapes_336.png` | "a single, solid black circle on a white background" | "a single white circle on a black background" |
| `test_image_shapes.png` | "a single, solid black circle on a white background" | "a single, solid black circle on a plain white background" |

The solid-colour fixture is read correctly once the ordering matches, which is what pins this defect. **The ordering fix alone is necessary but not sufficient:** the three-shape fixtures change but are still described wrong, so at least one further defect remains in this family's vision path and should be chased after this one lands rather than folded into it.

## Proposed Solution

1. Emit rows in merge-block-major order and inner features in `(dy, dx, c)` order, mirroring `convert_image_to_patches`:

```rust
for block_y in 0..h_patches / self.spatial_merge_size {
    for block_x in 0..w_patches / self.spatial_merge_size {
        for inner_y in 0..self.spatial_merge_size {
            for inner_x in 0..self.spatial_merge_size {
                // rows: (block_y, block_x, inner_y, inner_x)
                for dy in 0..self.patch_size {
                    for dx in 0..self.patch_size {
                        for c in 0..in_channels {
```

   `smart_resize` rounds both edges to `patch_size * spatial_merge_size`, so `h_patches` and `w_patches` are always exact multiples of `spatial_merge_size` and the block loops need no padding.

2. Replace the stale `unfold` comment with one naming this checkpoint's actual construction and linking https://huggingface.co/tencent/Youtu-VL-4B-Instruct/blob/main/image_processing_siglip2_fast.py (convert_image_to_patches).

3. Add a checkpoint-free unit test pinning the row order and the inner feature order for a grid with more than one block per edge, in the shape of `owned_and_mlx_paths_share_spatial_merge_grouped_patch_order`. A 2x2-block grid is the minimum that separates raster from block-major.

4. Validate on the real checkpoint at the three fixture sizes above and record the descriptions.

5. Once this lands, add `tests/youtu_vl_parity.rs` in the shape of `tests/qwen2_5_vl_parity.rs`: content-asserting `#[ignore]`d tests at 448 and 336 (both non-involution for this family, so they also guard #1600's window inverse) plus the 224 single-window control. That test file is written and was held back from #1603 because it cannot pass until this issue is fixed.

## Scope

**In scope:** the patch emission loop and its comment in `src/vision/processors/youtu_vl.rs`, its unit tests, and the real-checkpoint validation.

**Out of scope:** the window inverse (#1600). The processor's patch-count cap, which is a separate divergence filed on its own.

## Acceptance Criteria

- [x] `try_preprocess_with_spatial` emits merge-block-major rows with `(dy, dx, c)` inner features, matching `convert_image_to_patches`.
- [x] A checkpoint-free unit test separates block-major from raster on a grid with at least 2x2 blocks, and pins the inner feature order.
- [x] The 224x224 fixture is named as orange, not black, by the real checkpoint, with the before and after text recorded in the PR.
- [x] The two three-shape fixtures have their before and after recorded too. They changed and are still wrong, which the Evidence section above already predicted; the residual defect is filed as #1618 and is out of scope here.
- [x] `tests/youtu_vl_parity.rs` added in the shape of `tests/qwen2_5_vl_parity.rs`, carrying the 224 control (passing) and the two shape assertions written at the full correct answer and gated as known failing against #1618.
- [x] `cargo test --profile test-fast --features metal,accelerate --lib vision::processors::youtu_vl` (8 passed), `cargo clippy --lib --tests --features metal,accelerate -- -D warnings` and `cargo fmt --all -- --check` are green. The full `--workspace` gate was left to CI.

## Technical Considerations

Related: #1600 / #1603 (the window inverse in the same family, which this defect masks), #1596 / #1601 (the Qwen2.5-VL pair of defects that this family mirrors).
